sycl/plugins/level_zero/CMakeLists.txt

```diff
@@ -8,136 +8,87 @@ else()
-    COMPONENT level-zero-loader
-  )
+# Disable due to a bug https://github.com/oneapi-src/level-zero/issues/104
+set(CMAKE_INCLUDE_CURRENT_DIR OFF)
 
-  set(LEVEL_ZERO_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/level_zero_loader_install/include/
-	  CACHE INTERNAL "Path containing Level_Zero header files.")
-else()
-  file(GLOB LEVEL_ZERO_LIBRARY_SRC "${LEVEL_ZERO_LIBRARY}*")
-  get_filename_component(LEVEL_ZERO_LIB_NAME ${LEVEL_ZERO_LIBRARY} NAME)
-  add_custom_target(level-zero-loader
-      DEPENDS
-	  ${LLVM_LIBRARY_OUTPUT_INTDIR}/${LEVEL_ZERO_LIB_NAME}
-	  ${LLVM_BINARY_DIR}/include/sycl/level_zero
-  )
-  add_custom_command(
-      OUTPUT
-	  ${LLVM_LIBRARY_OUTPUT_INTDIR}/${LEVEL_ZERO_LIB_NAME}
-	  ${LLVM_BINARY_DIR}/include/sycl/level_zero
-      COMMENT
-	  "Copying Level Zero loader and headers"
-      COMMAND
-	  ${CMAKE_COMMAND} -E copy ${LEVEL_ZERO_LIBRARY_SRC} ${LLVM_LIBRARY_OUTPUT_INTDIR}
-      COMMAND
-	  ${CMAKE_COMMAND} -E copy_directory ${LEVEL_ZERO_INCLUDE_DIR} ${LLVM_BINARY_DIR}/include/sycl
-      DEPENDS
-	  ${LEVEL_ZERO_LIBRARY}
-          ${LEVEL_ZERO_INCLUDE_DIR}
-  )
-endif()
+message(STATUS "Will fetch Level Zero Loader from ${LEVEL_ZERO_LOADER_REPO}")
+include(FetchContent)
+FetchContent_Declare(level-zero-loader
+  GIT_REPOSITORY    ${LEVEL_ZERO_LOADER_REPO}
+  GIT_TAG           ${LEVEL_ZERO_LOADER_TAG}
+)
 
-list(APPEND SYCL_TOOLCHAIN_DEPLOY_COMPONENTS level-zero-loader level-zero-headers)
+FetchContent_MakeAvailable(level-zero-loader)
+FetchContent_GetProperties(level-zero-loader)
 
-include_directories("${LEVEL_ZERO_INCLUDE_DIR}")
+set(LEVEL_ZERO_INCLUDE_DIR
+  ${level-zero-loader_SOURCE_DIR}/include CACHE PATH "Path to Level Zero Headers")
 
 add_library (LevelZeroLoader-Headers INTERFACE)
-add_library (LevelZeroLoader::Headers ALIAS LevelZeroLoader-Headers)
 target_include_directories(LevelZeroLoader-Headers
   INTERFACE "${LEVEL_ZERO_INCLUDE_DIR}"
 )
 
 if (SYCL_ENABLE_XPTI_TRACING)
   set(XPTI_PROXY_SRC "${CMAKE_SOURCE_DIR}/../xpti/src/xpti_proxy.cpp")
+  set(XPTI_INCLUDE "${CMAKE_SOURCE_DIR}/../xpti/include")
+  set(XPIT_LIBS "${CMAKE_DL_LIBS}")
 endif()
 
+find_package(Python3 REQUIRED)
+
+add_custom_target(ze-api
+  COMMAND ${Python3_EXECUTABLE}
+  ${CMAKE_CURRENT_SOURCE_DIR}/ze_api_generator.py
+  ${LEVEL_ZERO_INCLUDE_DIR}/ze_api.h
+  BYPRODUCTS
+    ${CMAKE_CURRENT_BINARY_DIR}/ze_api.def
+  )
+
 find_package(Threads REQUIRED)
 
 add_sycl_plugin(level_zero
   SOURCES
-    "${sycl_inc_dir}/sycl/detail/pi.h"
-    "${CMAKE_CURRENT_SOURCE_DIR}/pi_level_zero.cpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/pi_level_zero.hpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/usm_allocator.cpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/usm_allocator.hpp"
-    "${CMAKE_CURRENT_SOURCE_DIR}/tracing.cpp"
+    # These are short-term shared with Unified Runtime
+    # The two plugins define a few things differrently so must
+    # be built separately. This difference is spelled in
+    # their "ur_bindings.hpp" files.
+    #
+    "ur_bindings.hpp"
```

Comment:
This is the only tricky change I would say. Basically some functions in "Unified Runtime", those that allocate new handles, act as a factory. The Level-Zero PI Plugin handle inherits from the Unified Runtime extending it with more functionality (still gradually all moving to Unified Runtime). Since this is a short-living solution think these "bindings" appeared to be good enough.